snuba/query/parser/functions.py

```diff
@@ -124,6 +132,26 @@ def parse_function(
         )
 
     name, args, alias = function_tuple
+
+    # If the first argument is a simple column name that refers to an array column
+    # (and we are not arrayJoining on that column, which would make it scalar again)
+    # we assume that the user actually means to check if any (or all) items in the
+    # array match the predicate, so we return an `any(x == value for x in array_column)`
+    # type expression. We assume that operators looking for a specific value (IN, =, LIKE)
+    # are looking for rows where any array value matches, and exclusionary operators
+    # (NOT IN, NOT LIKE, !=) are looking for rows where all elements match (eg. all NOT LIKE 'foo').
+    if name in FUNCTION_TO_OPERATOR:
+        if len(args) == 2 and args[0] in dataset_columns:
+            column = dataset_columns[args[0]]
+            if isinstance(column.type, Array) or (
+                isinstance(column.type, Materialized)
+                and isinstance(column.type.get_raw(), Array)
+            ):
+                if column.flattened != arrayjoin:
```

Comment:
You can have multiple arrayJoin columns in a query.
```
SELECT event_id, arrayJoin(tags.key), arrayJoin(contexts.key) FROM sentry_local;
```
Is a valid query for clickhouse and for snuba as well.
Not that I would like anybody to ever do that, but again, if we do not support that we need to reject queries that do that.

Reply:
I'll disallow it for now. Something that no one is Sentry is using, and if they need to we can address this then.